test/runnable/test18545.d

```diff
@@ -0,0 +1,18 @@
+module test18545;
+
+struct S {
+    int i;
+
+    @property const int get() { return 0; }
+
+    alias get this;
+}
+
+void main() {
+    const(S) s1 = S(42);
+
+    // this should reinterpret-cast, NOT call the implicit constructor with .get!
+    auto s2 = cast() s1;
+
```

Comment:
please add tests for `cast(immutable)` and `cast(const)` too.

Reply:
done: test now goes through the entire matrix of `cast([mutable, const, immutable]) [mutable, const, immutable]`. The problem occurs (and is fixed) with every cast between different constnesses.